### ISSUE TITLE
FR: Ignore code action

### ISSUE THREAD

**Issue by chrisgrieser** [kosayoda/nvim-lightbulb#51]
Would it maybe be possible to ignore certain code actions? Like, some LSP servers have a bunch of useful code actions, but also a bunch of useless code actions, and the lightbulb appears for both.

**Comment by kosayoda** [comment]
Hello! Does the `action_kinds` option do what you want? Pasting the relevant docs here:

```lua
-- Code action kinds to observe.
    -- To match all code actions, set to `nil`.
    -- Otherwise, set to a table of kinds.
    -- Example: { "quickfix", "refactor.rewrite" }
    -- See: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#codeActionKind
    action_kinds = nil,
```

**Comment by chrisgrieser** [comment]
Ah yes, that looks indeed looks je it could be it. 

Do you know how I can tell which type a particular code action has?

**Comment by kosayoda** [comment]
Hello again. I just added a `NvimLightbulb.debug()` function, can you let me know if this helps? Among other debug information it should show code actions at the location along with their type.

Example output:
<img width="466" alt="image" src="https://github.com/kosayoda/nvim-lightbulb/assets/41782385/90bdf660-2dff-4308-a0b7-6afeb7c3bd31">

**Comment by wookayin** [comment]
As said in https://github.com/kosayoda/nvim-lightbulb/issues/58#issuecomment-1870939846, allowing users to have a custom predicate or filter function `fun(lsp.CodeAction|lsp.Command): boolean` would be a better and general enough solution.

**Comment by kosayoda** [comment]
Hello! The `filter` feature is now implemented in [18c54dd](https://github.com/kosayoda/nvim-lightbulb/commit/18c54dd28d25b5bad7401f27fb73224f9ccac61a).

Thank you very much for your patience.
